### PR TITLE
fix!(hitbtc): hitbtc3 renamed to hitbtc

### PR DESCRIPTION
**Pull request by samgermain**
fixes: #18309

the hitbtc class now uses the v3 api, the hitbtc3 class remains as an alias

I'm also working on the websockets class in https://github.com/ccxt/ccxt/pull/18027

----------

### Typescript

```
2023-07-11T23:57:29.160Z
Node.js: v18.15.0
CCXT v4.0.8
hitbtc.fetchMarkets ()
2023-07-11T23:57:31.847Z iteration 0 passed in 343 ms

            id |              symbol |          base | quote | settle |   baseId | quoteId | settleId | type |  spot | margin |  swap | future | option | active | contract | linear | inverse |    taker |   maker | contractSize | expiry | expiryDatetime | strike | optionType | feeCurrency |                           precision |                                                                                              limits
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
       DUSKBTC |            DUSK/BTC |          DUSK |   BTC |        |     DUSK |     BTC |          | spot |  true |  false | false |  false |  false |   true |    false |        |         |   0.0025 |   0.001 |              |        |                |        |            |         BTC |        {"amount":0.01,"price":1e-9} |      {"leverage":{"min":1,"max":1},"amount":{"min":0.01},"price":{"min":1e-9},"cost":{"min":1e-11}}
       SNTUSDT |            SNT/USDT |           SNT |  USDT |        |      SNT |    USDT |          | spot |  true |  false | false |  false |  false |   true |    false |        |         |   0.0025 |   0.001 |              |        |                |        |            |        USDT |           {"amount":1,"price":1e-7} |          {"leverage":{"min":1,"max":1},"amount":{"min":1},"price":{"min":1e-7},"cost":{"min":1e-7}}
...
      CTSIUSDT |           CTSI/USDT |          CTSI |  USDT |        |     CTSI |    USDT |          | spot |  true |  false | false |  false |  false |   true |    false |        |         |   0.0025 |   0.001 |              |        |                |        |            |        USDT |     {"amount":0.01,"price":0.00001} |    {"leverage":{"min":1,"max":1},"amount":{"min":0.01},"price":{"min":0.00001},"cost":{"min":1e-7}}
        NEODAI |             NEO/DAI |           NEO |   DAI |        |      NEO |     DAI |          | spot |  true |  false | false |  false |  false |   true |    false |        |         |   0.0025 |   0.001 |              |        |                |        |            |         DAI |     {"amount":0.01,"price":0.00001} |    {"leverage":{"min":1,"max":1},"amount":{"min":0.01},"price":{"min":0.00001},"cost":{"min":1e-7}}
1567 objects
2023-07-11T23:57:31.847Z iteration 1 passed in 343 ms
```

```
2023-07-11T23:59:38.599Z
Node.js: v18.15.0
CCXT v4.0.8
hitbtc.fetchBalance ()
2023-07-11T23:59:41.462Z iteration 0 passed in 300 ms

{
  info: [
    {
      currency: 'BTC',
      available: '0.000010000000',
      reserved: '0',
      reserved_margin: '0',
      cross_margin_reserved: '0'
    },
    {
      currency: 'USDT',
      available: '0.736088192950',
      reserved: '0',
      reserved_margin: '0',
      cross_margin_reserved: '0'
    },
    {
      currency: 'USDC',
      available: '2.555903456391',
      reserved: '0',
      reserved_margin: '0',
      cross_margin_reserved: '0'
    }
  ],
  BTC: { free: 0.00001, used: 0, total: 0.00001 },
  USDT: { free: 0.73608819295, used: 0, total: 0.73608819295 },
  USDC: { free: 2.555903456391, used: 0, total: 2.555903456391 },
  free: { BTC: 0.00001, USDT: 0.73608819295, USDC: 2.555903456391 },
  used: { BTC: 0, USDT: 0, USDC: 0 },
  total: { BTC: 0.00001, USDT: 0.73608819295, USDC: 2.555903456391 }
}
2023-07-11T23:59:41.462Z iteration 1 passed in 300 ms
```

### Python

```
Python v3.11.3
CCXT v4.0.8
hitbtc.fetchMarkets()
[{'active': True,
  'base': 'DUSK',
  'baseId': 'DUSK',
  'contract': False,
  'contractSize': None,
  'expiry': None,
  'expiryDatetime': None,
  'feeCurrency': 'BTC',
  'future': False,
  'id': 'DUSKBTC',
  'info': {'base_currency': 'DUSK',
           'fee_currency': 'BTC',
           'make_rate': '0.001',
           'quantity_increment': '0.01',
           'quote_currency': 'BTC',
           'status': 'working',
           'take_rate': '0.0025',
           'tick_size': '0.000000001',
           'type': 'spot'},
  'inverse': None,
  'limits': {'amount': {'max': None, 'min': 0.01},
             'cost': {'max': None, 'min': 1e-11},
             'leverage': {'max': 1, 'min': 1.0},
             'price': {'max': None, 'min': 1e-09}},
  'linear': None,
  'maker': 0.001,
  'margin': False,
  'option': False,
  'optionType': None,
  'precision': {'amount': 0.01, 'price': 1e-09},
  'quote': 'BTC',
  'quoteId': 'BTC',
  'settle': None,
  'settleId': None,
  'spot': True,
  'strike': None,
  'swap': False,
  'symbol': 'DUSK/BTC',
  'taker': 0.0025,
  'type': 'spot'},
 ...
 {'active': True,
  'base': 'NEO',
  'baseId': 'NEO',
  'contract': False,
  'contractSize': None,
  'expiry': None,
  'expiryDatetime': None,
  'feeCurrency': 'DAI',
  'future': False,
  'id': 'NEODAI',
  'info': {'base_currency': 'NEO',
           'fee_currency': 'DAI',
           'make_rate': '0.001',
           'quantity_increment': '0.01',
           'quote_currency': 'DAI',
           'status': 'working',
           'take_rate': '0.0025',
           'tick_size': '0.00001',
           'type': 'spot'},
  'inverse': None,
  'limits': {'amount': {'max': None, 'min': 0.01},
             'cost': {'max': None, 'min': 1e-07},
             'leverage': {'max': 1, 'min': 1.0},
             'price': {'max': None, 'min': 1e-05}},
  'linear': None,
  'maker': 0.001,
  'margin': False,
  'option': False,
  'optionType': None,
  'precision': {'amount': 0.01, 'price': 1e-05},
  'quote': 'DAI',
  'quoteId': 'DAI',
  'settle': None,
  'settleId': None,
  'spot': True,
  'strike': None,
  'swap': False,
  'symbol': 'NEO/DAI',
  'taker': 0.0025,
  'type': 'spot'}]
```

```
Python v3.11.3
CCXT v4.0.8
hitbtc.fetchBalance()
{'BTC': {'free': 1e-05, 'total': 1e-05, 'used': 0.0},
 'USDC': {'free': 2.555903456391, 'total': 2.555903456391, 'used': 0.0},
 'USDT': {'free': 0.73608819295, 'total': 0.73608819295, 'used': 0.0},
 'free': {'BTC': 1e-05, 'USDC': 2.555903456391, 'USDT': 0.73608819295},
 'info': [{'available': '0.000010000000',
           'cross_margin_reserved': '0',
           'currency': 'BTC',
           'reserved': '0',
           'reserved_margin': '0'},
          {'available': '0.736088192950',
           'cross_margin_reserved': '0',
           'currency': 'USDT',
           'reserved': '0',
           'reserved_margin': '0'},
          {'available': '2.555903456391',
           'cross_margin_reserved': '0',
           'currency': 'USDC',
           'reserved': '0',
           'reserved_margin': '0'}],
 'total': {'BTC': 1e-05, 'USDC': 2.555903456391, 'USDT': 0.73608819295},
 'used': {'BTC': 0.0, 'USDC': 0.0, 'USDT': 0.0}}
```

### CODE REVIEW

File: ts/src/hitbtc.ts [hunk 1]
```diff
@@ -1,14 +1,10 @@
-
```
Comment (carlosmiei) [inline ts/src/hitbtc.ts:1]: @samgermain if this is a rename only why do we have two different files and so many changes?

Comment (samgermain) [reply ts/src/hitbtc.ts:1]: we already had hitbtc and hitbtc3. 

This PR moves all the code on the file hitbtc3.ts to hitbtc.ts, which removes all the code that was originally inside hitbtc.ts. Also all the code from hitbtc3.ts was removed and replaced with the few lines of code to make an alias exchange

Comment (ttodua) [reply ts/src/hitbtc.ts:1]: yes, Sam's edit of htbtc3 seems correct

**Comment by samgermain** [comment]
I'll need to redo this PR because there's a lot of conflicts now

**Comment by samgermain** [comment]
reopened in #18658

**Pull request closed without merging**